feature: micronaut test netty leak

sdelamo opened the pull request:
Adds feature for https://github.com/micronaut-projects/micronaut-test/pull/1291

sdelamo committed:
feature: micronaut test netty leak

yawkat commented on starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautTestNettyLeak.java:
```diff
@@ -0,0 +1,92 @@
+
+import java.util.Set;
+
+@Requires(property = "micronaut.starter.feature.micronaut-test-netty-leak.enabled", value = StringUtils.TRUE, defaultValue = StringUtils.TRUE)
```
i dont think this is useful enough for most people to become a default

sdelamo replied:
we add this `@Requires` for every feature.  It enables users to disable the feature.

sdelamo committed:
only add dependency if netty

yawkat commented:
I think we also need some code to enable extension detection for junit 5?

sdelamo committed:
more flexible assetion

The test sees two files:
- starter-web-netty
- test-suite-graal

It seems GraalVM test is picking the one in starter-web-netty instead of one test-suite-graal

sdelamo committed:
JunitPlatformPropertyProvider

sdelamo commented:
> I think we also need some code to enable extension detection for junit 5?

I created a `src/test/resources/junit-platform.properties` with  `junit.jupiter.extensions.autodetection.enabled=true`

https://docs.junit.org/current/user-guide/#extensions-registration-automatic-enabling

yawkat commented on starter-core/src/main/java/io/micronaut/starter/feature/test/MicronautTestNettyLeak.java:
```diff
@@ -0,0 +1,101 @@
+
+    @Override
+    public boolean shouldApply(ApplicationType applicationType, Options options, Set<Feature> selectedFeatures) {
+        return options.getTestFramework() == TestFramework.JUNIT || options.getTestFramework() == TestFramework.SPOCK;
```
it should not be on by default for these frameworks

sdelamo replied:
why not?

yawkat replied:
leak detection is too strict and not useful for most people

sdelamo replied:
but it is strict, in the sense of, of given false positives?

yawkat replied:
yes

sdelamo committed:
instance of test

sdelamo committed:
not a default feature

yawkat commented:
thanks!
